### ISSUE TITLE
Simple custom GRU without packed sequences

### ISSUE THREAD

**Issue by TParcollet** [espnet/espnet#1201]
Hi there!

I'm trying to use a custom RNN instead of the ones provided, and unfortunately it does not operate on packed sequences. 

As a first step, i'm just trying to reproduce the results with a simple GRU (torch.nn.GRU) that operates on unpacked sequences, but the PER on TIMIT explodes to 500%. Here is the implementation of this model:

<img width="871" alt="Capture d’écran 2019-09-17 à 15 38 33" src="https://user-images.githubusercontent.com/11910731/65046597-3d98f400-d961-11e9-9fc5-0b726d2f6d26.png">

The RNN class remains unaltered (encoders.py), due to the fact that I just change the call to GRU to GRU_cudnn. 

Titouan.

**Comment by b-flo** [comment]
Hi,

Shouldn't you sort `pad_packed_sequence` input by sequence length and unsort `pack_padded_sequence` to go back to the original ordering? I'm not sure about your input and what you do outside `forward()` but if we only consider this snippet the order is not the same.

**Comment by sw005320** [comment]
Did the training go well (validation loss/accuracy is in a similar range to the original LSTM)? I want to clarify the problem.

**Comment by TParcollet** [comment]
Hi!

Well, here is a new picture with some curves. Blue curves represents the model described above, while orange ones depict the same model but without padding and packing (it's basically the same than you standard GRU). 

<img width="1043" alt="Capture d’écran 2019-09-20 à 10 55 28" src="https://user-images.githubusercontent.com/11910731/65313768-38bc8600-db95-11e9-9f9c-aa03e82573ce.png">

I suppose that something goes wrong with the padding / unpacking step, but why? Maybe a mismatch with the ilens...

**Comment by TParcollet** [comment]
My bad, just had to specify batch_first=True.

**Comment by sw005320** [comment]
Good to know. Please let us know if your custom GRU works well.
